src/BeamAdapter/component/mapping/AdaptiveBeamMapping.inl

```diff
@@ -237,21 +263,41 @@ void AdaptiveBeamMapping< TIn, TOut>::apply(const MechanicalParams* mparams, Dat
     }
 
     AdvancedTimer::stepBegin("computeNewInterpolation");
-    for(unsigned int i=0; i<m_pointBeamDistribution.size(); i++)
+
+    // effective computation for each element in pointBeamDistribution
+    auto apply_impl = [&](const PosPointDefinition& pointBeamDistribution)
     {
-        PosPointDefinition pointBeamDistribution = m_pointBeamDistribution[i];
+        int i = &pointBeamDistribution - &m_pointBeamDistribution[0];
+
         Vec<3, InReal> pos;
-        const Vec3 localPos(0.,pointBeamDistribution.baryPoint[1],pointBeamDistribution.baryPoint[2]);
+        const Vec3 localPos(0., pointBeamDistribution.baryPoint[1], pointBeamDistribution.baryPoint[2]);
         l_adaptativebeamInterpolation->interpolatePointUsingSpline(pointBeamDistribution.beamId, pointBeamDistribution.baryPoint[0], localPos, in, pos, false, xtest);
 
-        if(m_isSubMapping)
+        if (m_isSubMapping)
         {
-            if(m_idPointSubMap.size()>0)
+            if (m_idPointSubMap.size() > 0)
                 out[m_idPointSubMap[i]] = pos;
         }
         else
             out[i] = pos;
+    };
+
+#if HAS_SUPPORT_STL_PARALLELISM
+    if (d_parallelMapping.getValue())
+    {
+        std::for_each(std::execution::par_unseq, m_pointBeamDistribution.begin(), m_pointBeamDistribution.end(), apply_impl);
     }
+    else // standard for-loop for each element (sequential, ordered)
+    {
+        std::for_each(std::execution::seq, m_pointBeamDistribution.begin(), m_pointBeamDistribution.end(), apply_impl);
+    }
+#else
+    for (const auto& p : m_pointBeamDistribution)
+    {
+        apply_impl(p);
+    }
+#endif // HAS_SUPPORT_STL_PARALLELISM
```

Comment:
```suggestion
#if HAS_SUPPORT_STL_PARALLELISM
    if (d_parallelMapping.getValue())
    {
        std::for_each(std::execution::par_unseq, m_pointBeamDistribution.begin(), m_pointBeamDistribution.end(), apply_impl);
    }
    else
#endif // HAS_SUPPORT_STL_PARALLELISM
for (const auto& p : m_pointBeamDistribution)
{
    apply_impl(p);
}

```
The sequential approach was repeated twice. This suggestion applies also to the other similar code